torch/nn/parallel/distributed.py

```diff
@@ -391,7 +391,7 @@ def _sync_params(self):
                                          self.broadcast_bucket_size)
             for tensors, module_params_data in zip(result[1:], self.modules_params_data[1:]):
                 for tensor, param_data in zip(tensors, module_params_data):
-                    param_data.set_(tensor)
```

Comment:
Never mind, actually this is a tensor.data.  I wonder how much performance degradation this will cause, since this is on the DDP critical path.

Reply:
Is there another way with the redesign to alias tensor storage?  I didn't really understand how to do that anymore.

Reply:
It would be great to measure how much performance degradation this incurs, and if the overhead is significant we can consider adding a `allow_tensor_metadata_change` flag to `.detach()`.